src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -131,28 +174,65 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string paramete
                     this.logger?.LogDebug("Running PowerShell rule engine");
                     evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
                 }
+               
+                // Recursively handle nested templates 
+                dynamic jsonTemplate = JsonConvert.DeserializeObject(template);
+                dynamic processedTemplateResources = templatejObject["resources"];
```

Comment:
ARM templates are not case sensitive, but the `JToken` class is.  This means using the `[]` indexing functionality to look for a property will fail if the template uses a different casing.  To avoid this, instead of using the `[]` indexing directly, use the `InsensitiveToken` extension method.

Reply:
I'm not sure I know how to use insensitivetoken to get properties. How does that work exactly?

Reply:
`InsensitiveToken` is defined at https://github.com/Azure/template-analyzer/blob/2c289296c7a5921389f1a18333ac0c3f90282931/src/Analyzer.Utilities/JTokenExtensions.cs#L54.  You can see example usages of it throughout the codebase, or look at examples in the tests: https://github.com/Azure/template-analyzer/blob/2c289296c7a5921389f1a18333ac0c3f90282931/src/Analyzer.Utilities.UnitTests/JTokenExtensionsTests.cs#L73#L104.

Basically, just call `InsensitiveToken` like it's a method in the `JToken` class, and pass the path you're looking for as a string.

The advantage of doing it that way is that the expected object will still be returned even if, in the JSON, it's written as "Resources".  In the current code, using `["resources"]`, nothing will be returned because the casing is different.